mirrord/layer/src/exec_utils.rs

```diff
@@ -279,21 +205,22 @@ pub(crate) unsafe extern "C" fn posix_spawn_detour(
     argv: *const *const c_char,
     envp: *const *const c_char,
 ) -> c_int {
-    match patch_sip_for_new_process(path, argv, envp) {
-        Success((new_path, new_argv, new_envp)) => {
-            let new_argv = new_argv.null_vec();
-            let new_envp = new_envp.null_vec();
+    let checked_envp = envp.checked_into();
 
-            FN_POSIX_SPAWN(
+    if let Detour::Success(modified_envp) = hooks::execve(checked_envp) {
+        match patch_sip_for_new_process(path, argv, modified_envp) {
```

Comment:
Same here for sip on if the `Detour` isn't `Success`, I think we should patch sip regardless of if we managed to change the envp

Reply:
Calling `patch_sip` first now, and if the `execve` preparation fails we still use the patched stuff.